Source/Rendering/SvgRenderer.cs

```diff
@@ -162,5 +178,75 @@ public static ISvgRenderer FromNull()
             var g = CreateGraphics(img);
             return new SvgRenderer(g, img);
         }
+
+        public void SetMask(Bitmap mask)
+        {
+            _mask = mask;
+        }
+
+        public Bitmap GetMask()
+        {
+            return _mask;
+        }
+
+        private void DrawMasked(Action<Graphics> drawAction, RectangleF bounds)
+        {
+            if (_mask == null)
+            {
+                drawAction(_innerGraphics);
+                return;
+            }
+
+            var renderedBounds = bounds.GetIntersection(this.RenderSize).Round();
+
+            if (renderedBounds.Width == 0 || renderedBounds.Height == 0)
+            {
+                return;
+            }
+
+            var buffer = new Bitmap(renderedBounds.Width, renderedBounds.Height, PixelFormat.Format32bppArgb);
+
+            var localTransform = new Matrix();
+            localTransform.Translate(-renderedBounds.X, -renderedBounds.Y);
+            localTransform.Multiply(this.Transform);
+
+            var bufferGraphics = Graphics.FromImage(buffer);
+
+            bufferGraphics.Transform = localTransform;
+            drawAction(bufferGraphics);
+
+            ApplyAlphaMask(buffer, _mask, renderedBounds);
+
+            var previousTransform = _innerGraphics.Transform;
+            _innerGraphics.Transform = new Matrix();
+            _innerGraphics.DrawImageUnscaled(buffer, new Point(renderedBounds.X, renderedBounds.Y));
+            _innerGraphics.Transform = previousTransform;
+        }
+
+        private void ApplyAlphaMask(Bitmap buffer, Bitmap mask, Rectangle renderedBounds)
+        {
+            var bufferData = buffer.LockBits(new Rectangle(0, 0, renderedBounds.Width, renderedBounds.Height), ImageLockMode.ReadWrite, PixelFormat.Format32bppArgb);
+            var maskData = mask.LockBits(renderedBounds, ImageLockMode.ReadOnly, PixelFormat.Format32bppArgb);
+
+            unsafe
```

Comment:
Is there implementation that doesn't use `unsafe` ?

Reply:
There are two I'm aware of - _GetPixel()_/_SetPixel()_ that's really slow or using the _Marshal_ class to copy the bitmap data but it uses a lot of memory.